src/GraphQL/Resolver/ImageAssetFieldResolver.php

```diff
@@ -6,52 +6,50 @@
  */
 namespace EzSystems\EzPlatformGraphQL\GraphQL\Resolver;
 
-use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
-use eZ\Publish\Core\FieldType\ImageAsset\AssetMapper;
-use EzSystems\EzPlatformGraphQL\GraphQL\DataLoader\ContentLoader;
 use EzSystems\EzPlatformGraphQL\GraphQL\Value\Field;
+use Ibexa\GraphQL\GraphQL\Mapper\ImageAssetMapperStrategyInterface;
 
 /**
  * @internal
  */
 class ImageAssetFieldResolver
 {
+    /* @var \Ibexa\Core\FieldType\ImageAsset\ImageAssetMapperStrategyInterface[] */
+    private $strategies;
+
     /**
-     * @var DomainContentResolver
-     */
-    private $domainContentResolver;
-    /**
-     * @var ContentLoader
-     */
-    private $contentLoader;
-    /**
-     * @var AssetMapper
+     * @param iterable<ImageAssetMapperStrategyInterface> $strategies
      */
-    private $assetMapper;
+    public function __construct(iterable $strategies)
+    {
+        foreach ($strategies as $strategy) {
+            if ($strategy instanceof ImageAssetMapperStrategyInterface) {
+                $this->addStrategy($strategy);
+            }
+        }
+    }
 
-    public function __construct(ContentLoader $contentLoader, DomainContentResolver $domainContentResolver, AssetMapper $assetMapper)
+    private function addStrategy(ImageAssetMapperStrategyInterface $strategy): void
```

Comment:
This method doesn't really bring value to the class. It's only used once and it's private. I'd recommend removing it or making it public (which could be beneficial in that case).